src/character.cpp

```diff
@@ -480,60 +480,39 @@ void Character::recalc_sight_limits()
     }
 }
 
-float Character::get_vision_threshold(int light_level) const {
-    // Bail out in extremely common case where character hs no special vision mode or
-    // it's too bright for nightvision to work.
-    if( vision_mode_cache.none() || light_level > LIGHT_AMBIENT_LIT ) {
-        return LIGHT_AMBIENT_LOW;
+static float threshold_for_range( float range )
+{
+    constexpr float epsilon = 0.01f;
+    return LIGHT_AMBIENT_MINIMAL / exp( range * LIGHT_TRANSPARENCY_OPEN_AIR ) - epsilon;
+}
+
+float Character::get_vision_threshold( float light_level ) const {
+    if( vision_mode_cache[DEBUG_NIGHTVISION] ) {
+        // Debug vision always works with absurdly little light.
+        return 0.01;
     }
+
     // As light_level goes from LIGHT_AMBIENT_MINIMAL to LIGHT_AMBIENT_LIT,
     // dimming goes from 1.0 to 2.0.
     const float dimming_from_light = 1.0 + (((float)light_level - LIGHT_AMBIENT_MINIMAL) /
                                             (LIGHT_AMBIENT_LIT - LIGHT_AMBIENT_MINIMAL));
-    float threshold = LIGHT_AMBIENT_LOW;
-
-    /**
-     * Consider vision modes in order of descending goodness until we get a hit.
-     * The values are based on expected sight distance in "total darkness", which is set to 3.7.
-     * The range is given by the formula distance = -log(threshold / light_level) / attenuation
-     * This is an upper limit, any smoke or similar should shorten the effective distance.
-     * The numbers here are hand-tuned to provide the desired ranges,
-     * would be nice to derive them with a constexpr function or similar instead.
-     */
-    if( vision_mode_cache[DEBUG_NIGHTVISION] ) {
-        // Debug vision always works with absurdly little light.
-        return 0.01;
-    } else if( vision_mode_cache[NV_GOGGLES] || vision_mode_cache[NIGHTVISION_3] ||
-               vision_mode_cache[FULL_ELFA_VISION] || vision_mode_cache[CEPH_VISION] ) {
-        if( vision_mode_cache[BIRD_EYE] ) {
-            // Bird eye adds one, so 13.
-            threshold = 1.9;
-        } else {
-            // Highest normal night vision is expected to provide sight out to 12 squares.
-            threshold = 1.99;
-        }
-    } else if( vision_mode_cache[ELFA_VISION] ) {
-        // Range 7.
-        threshold = 2.65;
+
+    float range = get_per() / 4.0f - encumb( bp_eyes ) / 10.0f;
```

Comment:
Should probably be.

But that still doesn't explain the 10 per thing.

Is 10 the threshold or just the first value tested that worked?